File: src/local.gwiasda.net/net.gwiasda.fima.reports/Model/CategoryReport.cs
namespace Net.Gwiasda.FiMa
{
    public class CategoryReport
    {
        public FinanceCategory Category { get; set; }
        public bool IsCost { get; set; }
        public decimal Sum { get; set; }
        public List<CategoryReport> ChildCategories { get; } = new List<CategoryReport>();
        public List<RecurringBooking> RecurringBookings { get; } = new List<RecurringBooking>();
        public List<Booking> Bookings { get; } = new List<Booking>();
    }
}

File: src/local.gwiasda.net/net.gwiasda.local.ui/ClientApp/src/app/fima/fima-reports/categoryReport.ts
import { FinanceCategory } from '../fima-categories/finance_category';
import { Booking } from '../fima-booking/booking';

export class CategoryReport {
  category: FinanceCategory = new FinanceCategory();
  isCost: boolean = false;
  sum: number = 0;
  childCategories: CategoryReport[] = [];
  bookings: Booking[] = [];
}
